### DISCUSSION TITLE
String parsing into Function

### DISCUSSION THREAD

**Discussion by amoyyy** [vlang/v#11933] [Ideas]
Is it possible to fulfill codes like below? Thanks for any advice provided.
```
fn fn_generator(expr string) fn(f64) f64 {
	return fn (a f64) f64 {
                // could 'fn' be auto-generated by parsing 'expr'? 
		return 6.0*a*a+6.0
	}
}

fn main(){
	mut var := 5.0
	expr := "6.0*x^2+6.0"
	op = fn_generator(expr)
	println(op(var))
}
```

**Comment by amoyyy** [comment]
Some algorithms like 'precedence climb', 'ShuntingYard', 'TDOP', .etc. have the functionality actually. But I expect an implemement which directly parses 'string' into 'native V functions' to acquire runtime efficiency. I've tried to find some tips in v.ast codes but I'm not familiar with compiler mechanism. I'm grateful if anyone knows how to achieve the goal. Hope for advice!

**Comment by JalonSolov** [comment]
What you're thinking of is similar to the `eval` functionality in other (usually interpreted, not compiled) languages.  V doesn't have that.

Something that _might_ work, though... use the hotload functionality.  Write a routine that will take a string of code, write it to a file, and have that marked as a `[live]` routine in V so that V will take care of compiling & reloading what was just written as a usable function.

That's the only way I can think of that might get what you want, without massive changes to V.
